src/python/pants/backend/python/util_rules/interpreter_constraints_test.py

```diff
@@ -293,7 +293,7 @@ def test_group_field_sets_by_constraints_with_unsorted_inputs() -> None:
     "constraints,expected",
     (
         (["==2.7.*"], "==2.7.*"),
-        (["==2.7.*", ">=3.6,!=3.6.1"], "==2.7.* || !=3.6.1,>=3.6"),
+        (["==2.7.*", ">=3.6,!=3.6.1"], "!=3.6.1,>=3.6 || ==2.7.*"),
```

Comment:
Original version of this was affected by the incorrect sorting behaviour in `InterpreterConstraints()`. This should be equivalent.